dipy/reconst/rumba.py

```diff
@@ -0,0 +1,815 @@
+        int.
+    x : float or ndarray
+        value or array of values with which to compute ratio
+
+    Returns
+    -------
+    y : float or ndarray
+        Result of ratio computation.
+
+    References
+    ----------
+    .. [1] W. Gautschi and J. Slavik, “On the computation of modified Bessel function ratios,”
+           Math. Comp., vol. 32, no. 143, pp. 865–875, 1978, doi: 10.1090/S0025-5718-1978-0470267-9
+    '''
+
+    y = x / ((2*n + x) - (2*x * (n + 1/2) / (2*n + 1 + 2*x - (2*x * (n + 3/2) / (
+        2*n + 2 + 2*x - (2*x * (n + 5/2) / (2*n + 3 + 2*x)))))))
+
+    return y
+
+
+def generate_kernel(gtab, sphere, lambda1=1.7e-3, lambda2=0.2e-3, lambda_csf=3.0e-3):
+    '''
+    Generate deconvolution kernel
+
+    Compute kernel mapping volume fractions of white matter fiber populations (oriented along each
+    vertex of a sphere) and cerebrospinal fluid (CSF) to a diffusion weighted signal.
+
+    Parameters
+    ----------
+    gtab : GradientTable
+    sphere : Sphere
+        sphere with which to sample discrete fiber orientations in order to construct kernel
+    lambda1 : float, optional
+        white matter diffusivity parallel to fiber axis (first DTI eigenvalue). Default: 1.7e-3
+    lambda2 : float
+        white matter diffusivity perpendicular to fiber axis (second/third DTI eigenvalues are
+        assumed equal). Default: 0.2e-3
+    lambda_csf : float
+        Mean diffusivity for CSF. If `None`, then CSF kernel column set to all zeroes.
+        Default: 3.0e-3
+
+    Returns
+    -------
+    kernel : 2d ndarray (N, M)
+        computed kernel; can be multiplied with a vector consisting of volume fractions for each of
+        M-1 fiber populations and the fraction of CSF to produce a diffusion weighted signal.
+    '''
+
+    # Coordinates of sphere vertices
+    _, theta, phi = cart2sphere(
+        sphere.x,
+        sphere.y,
+        sphere.z
+    )
+
+    n_grad = len(gtab.gradients)  # number of gradient directions
+    n_wm_comp = len(theta)  # number of fiber populations
+    n_comp = n_wm_comp + 1  # plus compartment for CSF
+
+    kernel = np.zeros((n_grad, n_comp))
+
+    S0 = 1  # S0 assumed to be 1
+    fi = 100  # volume fraction assumed to be 100%
+
+    # White matter components
+    for i in range(n_wm_comp):
+        # Convert angles to degrees
+        angles = [theta[i] * 180 / np.pi, phi[i] * 180 / np.pi]
+
+        # Signal generated by WM-fiber for each gradient direction
+        S, _ = multi_tensor(gtab, np.array([[lambda1, lambda2, lambda2]]),
+                            S0, [angles], [fi], None)
+        kernel[:, i] = S
+
+    # CSF component
+    if lambda_csf is None:
+        S_csf = np.zeros((n_grad))
+    else:
+        S_csf, _ = multi_tensor(gtab, np.array([[lambda_csf, lambda_csf, lambda_csf]]),
+                                S0, [[0, 0]], [100], None)
+
+    kernel[:, n_wm_comp] = S_csf
+
+    return kernel
+
+
+def combine_odf_csf(odf, f_csf):
```

Comment:
I think this function can be moved inside the RumbaFit object, and behave in the same way as `odf(.)` and `f_csf(.)` (call `_fit(.)` if `None`)